src/System.CommandLine.Suggest.Tests/SuggestionShellScriptHandlerTest.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Xunit;
+
+namespace System.CommandLine.Suggest.Tests
+{
+    public class SuggestionShellScriptHandlerTest
+    {
+        [Fact]
+        public void When_shellName_does_not_support_it_should_print_error()
+        {
+            var console = new TestConsole();
+            Action a = () => SuggestionShellScriptHandler.Handle(console, "notsupported");
+            a.Should().Throw<SuggestionShellScriptException>().And.Message.Should()
```

Comment:
This shows a stack trace in stderr, no?

Reply:
No, it's better to avoid using an exception for input validation or control flow. Writing the error to stderr is preferable.

Reply:
I don't think if an exception go all the way up is considered control flow, catch it or show up in happy path is. Exception is a good abstraction for when something is wrong and the program cannot resolve by itself. Stop all and here is the reason why. It is especially useful for command line app, there is nothing to retry or revert.